library/src/test/java/com/bumptech/glide/GlideTest.java

```diff
@@ -547,6 +552,44 @@ public void testClone() throws IOException {
     verify(secondTarget).onResourceReady(notNull(Drawable.class), isA(Transition.class));
   }
 
+  @Test
+  public void testSetModulesEnabledTrue() throws Exception {
+    // teaDown glide instance first so we have a clean slate and not the Glide.get() call in setUp
+    Glide.tearDown();
+
+    // This is the default, so it should be a no-op
+    Glide.setModulesEnabled(true);
+
+    Glide glide = Glide.get(getContext());
+    List<ModelLoader<GlideUrl, ?>> modelLoaders =
+        glide.getRegistry().getModelLoaders(mock(GlideUrl.class));
+
+    // With modules enabled, SetupModule below should have been initialized,
+    // so the default modelLoader should NOT be configured
+    assertEquals(1, modelLoaders.size());
+    assertFalse(modelLoaders.get(0) instanceof HttpGlideUrlLoader);
+  }
+
+  @Test
+  public void testSetModulesEnabledFalse() throws Exception {
+    // teaDown glide instance first so we have a clean slate and not the Glide.get() call in setUp
+    Glide.tearDown();
+
+    Glide.setModulesEnabled(false);
+    try {
+      Glide glide = Glide.get(getContext());
+      List<ModelLoader<GlideUrl, ?>> modelLoaders =
+          glide.getRegistry().getModelLoaders(mock(GlideUrl.class));
+
+      // With modules disabled, SetupModule below should not have been initialized,
+      // so the default modelLoader should remain
+      assertEquals(1, modelLoaders.size());
+      assertTrue(modelLoaders.get(0) instanceof HttpGlideUrlLoader);
+    } finally {
+      Glide.setModulesEnabled(true);
```

Comment:
I think this should be part of Glide.tearDown, it resets Glide to pre-initialized state, which IMO includes both static variables. (It should probably even reset ViewTarget.tagId)